Notes.md

```diff
@@ -0,0 +1,10 @@
+1. Dereference : Grabbing the underlying value that is located in a particular address
+  * int x  = 10; 
+    int *p = &x;  // address of x
+
+    printf("%d\n", *p) // deference the pointer and grab 10 from x 
+                       // which p is pointing to. *p will print 10. 
+
+2. Heap - is where all your allocated memory variables live and variables can be passed from one function to the next. 
```

Comment:
While it's true that variables can be passed from one function to another (either by reference or by value), that fact doesn't actually have anything to do with the heap.